File: netlify/functions/ingest_trigger.py
import json

CORS_HEADERS = {
    "Access-Control-Allow-Origin": "*",
    "Access-Control-Allow-Headers": "Content-Type",
    "Access-Control-Allow-Methods": "GET, POST, OPTIONS",
    "Content-Type": "application/json",
}


def handler(event, context):
    method = event.get("httpMethod", "GET")

    if method == "OPTIONS":
        return {"statusCode": 200, "headers": CORS_HEADERS, "body": ""}

    return {
        "statusCode": 200,
        "headers": CORS_HEADERS,
        "body": json.dumps(
            {
                "status": "ingestion_is_local",
                "message": "Run python scripts/ingest_documents.py <path_to_pdf> "
                "to ingest documents.",
            }
        ),
    }

File: netlify/functions/ingest_trigger.js
exports.handler = async () => {
  return {
    statusCode: 200,
    headers: { 'Access-Control-Allow-Origin': '*' },
    body: JSON.stringify({
      status: 'ingestion_is_local',
      message:
        'Run python scripts/ingest_documents.py <path_to_pdf> to ingest documents.',
    }),
  };
};
